torch/nested/__init__.py

```diff
@@ -410,3 +412,38 @@ def nested_tensor_from_jagged(
 
     return nested_view_from_values_offsets_lengths(
         values, offsets, lengths, ragged_idx=jagged_dim, min_seqlen=min_seqlen, max_seqlen=max_seqlen)
+
+def masked_select(tensor: Tensor, mask: Tensor) -> Tensor:
+    if tensor.layout != torch.strided:
+        raise RuntimeError(
+            f"torch.nested.masked_select requires a strided tensor, given {tensor.layout}"
+        )
+
+    if mask.layout != torch.strided:
+        raise RuntimeError(
+            f"torch.nested.masked_select requires a strided mask, given: {mask.layout}"
+        )
+
+    if not is_expandable_to(mask.shape, tensor.shape):
+        raise RuntimeError(
+            f"Mask shape {mask.shape} is not broadcastable to tensor shape {tensor.shape}"
+        )
+
+    expanded_mask = mask.expand(tensor.shape)
+
+    res_values = tensor.masked_select(mask)
```

Comment:
I see that we have a manual check for broadcasting above. Since masked_select should already does some kind of broadcasting check today, does that already fully cover this case?

Reply:
So I can flip the masked_select and expand ops here but we still need to know the expanded_mask in order to compute the lengths/offsets